crates/algokit_utils/src/transactions/composer.rs

```diff
@@ -1948,22 +1948,23 @@ impl Composer {
 
         // Group transactions by signer
         let mut transactions = Vec::new();
-        let mut signer_groups: HashMap<*const dyn TransactionSigner, Vec<usize>> = HashMap::new();
+        let mut signer_groups: Vec<(Arc<dyn TransactionSigner>, Vec<usize>)> = Vec::new();
+
         for (group_index, txn_with_signer) in transactions_with_signers.iter().enumerate() {
-            let signer_ptr = Arc::as_ptr(&txn_with_signer.signer);
```

Comment:
I want to point out that raw pointers in an async method makes it not thread safe (not Send + Sync) which we need for uniffi.